trio/tests/test_abc.py

```diff
@@ -19,3 +19,24 @@ class MyAR(tabc.AsyncResource):
         assert myar.record == []
 
     assert myar.record == ["ac"]
+
+
+def test_abc_generics():
```

Comment:
It'd be nice to have a comment on this explaining why we thought such an odd-looking test is useful :-)